lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +43,19 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or text == '':
+            return None
+
+        list_text = text.split()
+        str_text = self._end_of_word_token.join(list_text)
+        new_str = ''
+        for token in str_text:
+            if token.isalpha() or token == '_':
+                new_str += token
+        new_str += '_'
+        if '__' in new_str:
+            copy_str = new_str.replace('__', '_')
+            return tuple([token.lower() for token in copy_str])
```

Comment:
Convert sting to lower case  and than just list. Without any loops

Reply:
done